### PR TITLE
Split deployment

### PR DESCRIPTION
**Pull request by ehelms**
This makes a few changes. The biggest change is splitting the playbook to focus on separating concerns from the get go. The all-in-one is easily still supported by defining each group as the same host. This does however provide a framework for split deployment thinking from the outset and shows dependencies for each  'service'. This also moves away from playbooks placed on the system itself to perform operations towards a model of a roole that is dedicated to the task. This I think accomplishes the ability to re-run the action while simplifying the design.

### CODE REVIEW

File: roles/foreman_postgresql/tasks/main.yml [hunk 1]
```diff
@@ -21,10 +21,18 @@
     owner: postgres
     group: postgres
 
+- name: Set listen addresses to *
```
Comment (ares) [inline roles/foreman_postgresql/tasks/main.yml:24]: Should we make this optional? For all-in-one, we may not want to listen everywhere.

File: roles/foreman_postgresql/templates/pg_hba.conf.j2 [hunk 2]
```diff
@@ -5,5 +5,9 @@ local   all             all                                     ident
 
 # IPv4 local connections:
 host    all             all             127.0.0.1/32            md5
+
+# IPv4 local connections:
+host    all             all             0.0.0.0/0               md5
```
Comment (ares) [inline roles/foreman_postgresql/templates/pg_hba.conf.j2:10]: Seems to duplicate lines above

File: roles/foreman_setup/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+foreman_setup_foreman_host: "{{ ansible_fqdn }}"
+foreman_setup_foreman_url: "https://{{ foreman_setup_foreman_host }}/api/v2/hosts"
+foreman_setup_data:
+  host:
+    name: "{{ ansible_fqdn }}"
+    location_id: 2
+    organization_id: 1
```
Comment (ares) [inline roles/foreman_setup/defaults/main.yml:8]: This is not great, we've seen it changed recently and then switched back again.

**Review by ares — CHANGES_REQUESTED** [review]
In general I like the change. Could we add a real inventory that would work for all-in-one with localhost? Until we have tooling for running the ansible.

File: roles/foreman_postgresql/tasks/main.yml (hunk 1 above)
Comment (ehelms) [reply roles/foreman_postgresql/tasks/main.yml:24]: How would you feel about leaving this and filing an issue or todo to support running on localhost?

File: roles/foreman_postgresql/templates/pg_hba.conf.j2 (hunk 2 above)
Comment (ehelms) [reply roles/foreman_postgresql/templates/pg_hba.conf.j2:10]: I will update the comment

File: roles/foreman_setup/defaults/main.yml (hunk 3 above)
Comment (ehelms) [reply roles/foreman_setup/defaults/main.yml:8]: Fair point. Registration is hard. Let's make an issue to resolve that since this already exists.

**Comment by ehelms** [comment]
Pushed some updates to fix running on a single remote host and on localhost by updating inventory.sample to support that use case.

**Comment by ares** [comment]
Thank you @ehelms, very much appreciated! Sorry for the delay, you deserve a :beer: and I'm happy I can buy you one right now.

**Pull request merged**